tiledb/api/tests/array/sync.stderr

```diff
@@ -0,0 +1,51 @@
+error[E0277]: `Rc<context::RawContext>` cannot be shared between threads safely
```

Comment:
Did you mean to check this in?

Reply:
Seems a little strange for me to assert on compiler output. Is the error format guaranteed to be stable going forward?

Reply:
Maybe we can simply assert it fails to compile.

Reply:
> Did you mean to check this in?

Yep

> Seems a little strange for me to assert on compiler output. Is the error format guaranteed to be stable going forward?

I don't imagine it would drift too much but I agree we should not assume it won't change.  The `trybuild` crate makes it easy to recapture (run with `TRYBUILD=1`).

> Maybe we can simply assert it fails to compile.

That is the goal but I think just "it fails to compile" is far too broad.  Syntax error is also a failure to compile but doesn't tell me what I'm looking for and possibly hides a success.  I'd be happy just checking the error number, but the other option to do that using doc tests did not work for some reason.  I'm fine with the extra verbose output that we might have to recapture every once in a while as a tradeoff.

I'd definitely love to use some kind of regexp or something if `trybuild` had the means...

Reply:
The rest of that paragraph seems like an *endorsement* rather than a reason not to capture the compiler errors.

The beginning of the section seems more on point:
> When it comes to compile-fail tests, write tests for anything for which you care to find out when there are changes in the user-facing compiler output. As a negative example, please don’t write compile-fail tests simply calling all of your public APIs with arguments of the wrong type; there would be no benefit.

This test is closer to "calling a public API with the wrong type" than "changes in user-facing compiler output".

I want to test that `Array` is not `Sync`.  For some reason the doc test `compile_fail,E0277` syntax did not work - it correctly checked that compilation failed but the error number was not used for some reason.  I would like to have something stronger than "fails to compile" but not as sensitive "if the compiler changes the text of its error message then the test fails and need to evaluate recapturing", but I haven't found an option for that.  And I think the over-sensitivity is the better choice.

Reply:
> I want to test that `Array` is not `Sync`. For some reason the doc test `compile_fail,E0277` syntax did not work -

As recently as a year ago, that required cargo nightly to work:

> ,E0515 indicates the error number you expect, which avoids any other errors that would make the test false negative. (This error number check will be ignored if cargo is not running as nightly)

via https://stackoverflow.com/questions/74989492/how-to-assert-that-code-wont-compile-in-a-test